test/e2e/create_test.go

```diff
@@ -175,17 +175,68 @@ func (tc *testContext) testMachineConfiguration(t *testing.T) {
 	}
 	_, err := tc.createWindowsMachineSet(gc.numberOfMachineNodes, false)
 	require.NoError(t, err, "failed to create Windows MachineSet")
+	machineCreationTime := time.Now()
 
 	t.Run("Machine configuration while private key change", tc.testMachineConfigurationWhilePrivateKeyChange)
 
 	machines, err := tc.waitForWindowsMachines(int(gc.numberOfMachineNodes), "Provisioned", false)
 	require.NoError(t, err, "error waiting for Windows Machines to be provisioned")
+	machineProvisionedTime := time.Now()
+	var addresses []string
+	for _, m := range machines.Items {
+		addr, err := controllers.GetAddress(m.Status.Addresses)
+		require.NoError(t, err)
+		addresses = append(addresses, addr)
+	}
+	err = tc.waitForSSHAvailable(addresses)
+	require.NoError(t, err)
+	sshAvailableTime := time.Now()
 	err = tc.waitForConfiguredWindowsNodes(gc.numberOfMachineNodes, false, false)
 	assert.NoError(t, err, "Windows node creation failed")
+
+	// Save timings as an artifact for performance measuring purposes
+	configTimes := configurationTimes{
+		NodeType:           "machine",
+		NodeCount:          int(gc.numberOfMachineNodes),
+		MachineProvisioned: machineProvisionedTime.Sub(machineCreationTime).Round(time.Second).String(),
+		SSHAvailable:       sshAvailableTime.Sub(machineProvisionedTime).Round(time.Second).String(),
+		NodeConfigured:     time.Now().Sub(sshAvailableTime).Round(time.Second).String(),
+	}
+	times, err := json.Marshal(configTimes)
+	require.NoError(t, err)
+	times = append(times, '\n')
+	fd, err := os.OpenFile(filepath.Join(tc.artifactDir, "config-time.json"), os.O_WRONLY|os.O_CREATE|os.O_APPEND, os.ModePerm)
```

Comment:
could you add sample of the config-time.json? would be nice to see the resulting content in the description of the PR.